external-crates/move/crates/move-compiler/src/typing/macro_expand.rs

```diff
@@ -0,0 +1,955 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
I really appreciate how direct this code is for a macro expander -- lots of clarity for something so complex. I do have a thought about combining the traversals, but we can look into those sorts of optimizations down the road.